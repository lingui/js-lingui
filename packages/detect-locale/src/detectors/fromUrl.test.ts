import { fromUrl } from ".."

describe("fromUrl", () => {
  it("parses correctly the param after an &", () => {
    const mock: Partial<Location> = {
      search: "?some_param=value&lang=en",
    }
    const locale = fromUrl("lang", mock)
    expect(locale).toEqual("en")
  })

  it("parses correctly the lang after an ?", () => {
    const mock: Partial<Location> = {
      search: "?lang=es",
    }
    const locale = fromUrl("lang", mock)
    expect(locale).toEqual("es")
  })

  it("parses correctly the lang with multiple params", () => {
    const mock: Partial<Location> = {
      search: "?lang=cs&some_other=other&otheR_params=232",
    }
    const locale = fromUrl("lang", mock)
    expect(locale).toEqual("cs")
  })

  it("throws an error if parameter is not passed", () => {
    const mock: Partial<Location> = {
      search: "?lang=cs&some_other=other&otheR_params=232",
    }
    expect(() => {
      fromUrl(undefined, mock)
    }).toThrowError()
    expect(() => {
      fromUrl(undefined, mock)
    }).toThrowErrorMatchingSnapshot()
  })
})
