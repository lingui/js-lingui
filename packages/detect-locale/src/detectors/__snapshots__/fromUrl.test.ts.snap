// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fromUrl throws an error if parameter is not passed 1`] = `"fromUrl parameter is required"`;
